include/power_grid_model/component/voltage_sensor.hpp

```diff
@@ -91,10 +91,18 @@ class VoltageSensor : public GenericVoltageSensor {
     RealValue<sym> u_measured_;
     RealValue<sym> u_angle_measured_;
 
+    bool has_angle() const {
+        if constexpr (sym) {
+            return !is_nan(u_angle_measured_);
+        }
+        else {
+            return !u_angle_measured_.isNaN().any();
+        }
+    }
+
     SensorCalcParam<true> sym_calc_param() const final {
-        bool const has_angle = !is_nan(u_angle_measured_);
         double u_variance = u_sigma_ * u_sigma_;
-        if (has_angle) {
+        if (has_angle()) {
```

Comment:
Since no unit test crashed when you changed this. Can you make a unit test that covers this behaviour?